### DISCUSSION TITLE
setError - outside declared field type

### DISCUSSION THREAD

**Discussion by tanetrail** [react-hook-form/react-hook-form#3926] [Q&A]
trying to set the error for outside declared field type and getting syntax error "**Argument of type 'string' is not assignable to parameter of type 'never'**"

tried this https://www.carlrippon.com/react-hook-form-server-validation/ but still same issue. only difference i see is, we cannot define the form fields upfront (ex: useForm<FormData>())

**Comment by bluebill1049** [comment]
https://react-hook-form.com/api#setError

> Note: for TypeScript users, please consider this approach for setting server-side error or outside declared field type.

maybe take a look at the example above.

**Comment by david-vendel** [comment]
Hello @tanetrail, I managed to fix that error getting setError from useFormContext, not from useForm. (I use useFormContext in my form).
